scripts/launch-s0-instance.sh

```diff
@@ -27,6 +27,7 @@ policy_name="s0-permissions"
 db_mount=/mnt/elastio
 unit_file=/etc/systemd/user/s0.service
 service=s0
+instance_market_options="--instance-market-options MarketType=spot,SpotOptions={SpotInstanceType=persistent,InstanceInterruptionBehavior=stop}"
```

Comment:
I'm not sure this is the right interruption behavior.  Shouldn't it terminate on interruption?  Did you test this with an instance running long enough that spot terminates it?

Reply:
Initially it was `--instance-market-options MarketType=spot,SpotOptions={SpotInstanceType=persistent}` The default `InstanceInterruptionBehavior` is `terminate`. But in this case command `aws ec2 run-instances` is failing with the
```
An error occurred (InvalidParameterCombination) when calling the RunInstances operation: The terminate InstanceInterruptionBehavior is not supported when requestType is set to persistent.
```
The  `terminate` works with the `SpotInstanceType` equal to `one-time` and dosn't work with the `persistent`.

Reply:
> Did you test this with an instance running long enough that spot terminates it?

No

Reply:
> A persistent Spot Instance request remains active until it expires or you cancel it, even if the request is fulfilled. If the Spot price exceeds your maximum price or capacity is not available, your Spot Instance is interrupted. After your instance is interrupted, when your maximum price exceeds the Spot price or capacity becomes available again, the Spot Instance is started if stopped or resumed if hibernated. You can stop a Spot Instance and start it again if capacity is available and your maximum price exceeds the current Spot price. If the Spot Instance is terminated (irrespective of whether the Spot Instance is in a stopped or running state), the Spot Instance request is opened again and Amazon EC2 launches a new Spot Instance. For more information, see Stop a Spot Instance, Start a Spot Instance, and Terminate a Spot Instance. 

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html

The only thing, I didn't set a `MaxPrice` value in the `SpotOptions`.

Reply:
Shouldn't we use `one-time` instead of `persistent`?

Reply:
To be honest, I don't know.

`one-time` will run until spot terminates it, and then it's gone.  `persistent` in theory means the spot request will remain outstanding and spot will try to re-launch the instance again on different compute.  that second option seems better in principle.  but how will it actually work?  will the re-launched instance work correctly, or is there some glitch there?

> The only thing, I didn't set a MaxPrice value in the SpotOptions.

The default is to use the on-demand price, which is a reasonable default.  No need to set it explicitly.

> No

Please do, or ask Maryna or Rybitskiy to do it for you.

Reply:
@acv02614 and @VladasZ already have two instances started using the script from the branch. So, we'll see.
The issue that we are already saw, `aws ec2 run-instances` reports an error, that there is no capacity and 2 retries are failed. And as result, the script returns an error. But a started spot instance appears in the list of the instances. That's the issue on the start. So, 2 instances are running, and we'll see, how they restart.

Reply:
@anelson the next behaviour has been observed:
- instances are rebooted time to time. how we figured out that? by appearance of the bug when ec2-user doesn't own database dir in /mnt after reboot.
- it's impossible to terminate instance via aws console gui. it creates a new one until the spot request is not cancelled.

So, for the dev's purposes is much better to use `one-time` spot instances with the instead of the `terminate` interruption behaviour. 
Also, for the provisioner's usage, the same `one-time` with `terminate` spot instance seems to be better. The provisioner service should just create a new instance if one is terminated.

But I'm not sure what's better for the early customers right now.

Reply:
It does seem that one-time instances with `terminate` behavior are better for now.

Reply:
Changed it to use `one-time` instances with `terminate` behavior.

Reply:
The only thing, now trying to check my changes and got eleven errors:
```
An error occurred (InsufficientInstanceCapacity) when calling the RunInstances operation (reached max retries: 2): There is no Spot capacity available that matches your request.

```

Reply:
Well, 15th attempt was successful ) it works